railties/lib/rails/generators/app_base.rb

```diff
@@ -298,6 +298,20 @@ def rails_version_specifier(gem_version = Rails.gem_version)
         end
       end
 
+      # This "npm-ifies" the current version number
+      # With npm, versions such as "5.0.0.rc1" or "5.0.0.beta1.1" are not compliant with its
+      # versioning system, so they must be transformed to "5.0.0-rc1" and "5.0.0-beta1-1" respectively.
+
+      # "5.0.1"     --> "5.0.1"
+      # "5.0.1.1"   --> "5.0.1-1" *
+      # "5.0.0.rc1" --> "5.0.0-rc1"
+      #
+      # * This makes it a prerelease. That's bad, but we haven't come up with
+      # a better solution at the moment.
+      def npm_version
+        Rails.version.gsub(/\./).with_index { |s, i| i >= 2 ? "-" : s }
```

Comment:
If we want to support `--main`, I think we need to do something like here:

https://github.com/rails/rails/blob/0f5201a756399911c3ce00f83047a1598240540a/railties/lib/rails/generators/app_base.rb#L268-L284

We may also want to support these other tags too :)

Reply:
this'll get us started https://github.com/rails/rails/pull/42263/commits/b5593614aed68d311c39b4c97fd5103f62e7ffa4